lib/components/field-key/list.vue

```diff
@@ -188,41 +224,56 @@ export default {
       });
       this.enabledPopoverLinks.add(index);
     },
-    showPopover(popoverLink) {
-      this.enablePopover(popoverLink);
-      popoverLink.popover('show');
-      this.popoverLink = popoverLink;
+    showPopover($popoverLink) {
+      this.enablePopover($popoverLink);
+      $popoverLink.popover('show');
+      this.popoverLink = $popoverLink.get(0);
     },
     hidePopover() {
       if (this.popoverLink == null) return;
-      this.popoverLink.popover('hide');
+      $(this.popoverLink).popover('hide');
       this.popoverLink = null;
     },
-    popoverContainsElement(element) {
-      if (this.popoverLink == null) return false;
-      const popover = $('#field-key-list-popover-content').closest('.popover');
-      return element === popover[0] || $.contains(popover[0], element);
+    elementIsOutsidePopover(element) {
+      if (this.popoverLink == null) return true;
+      const popover = $('#field-key-list-popover-content').closest('.popover')[0];
+      return !(element === popover || $.contains(popover, element));
     },
     togglePopovers(event) {
-      const popoverLink = $(event.target).closest('.field-key-list-popover-link');
-      if (popoverLink.length !== 0) {
+      const $popoverLink = $(event.target).closest('.field-key-list-popover-link');
+      if ($popoverLink.length !== 0) {
         // true if the user clicked on the link whose popover is currently shown
         // and false if not.
         const samePopover = this.popoverLink != null &&
-          popoverLink[0] === this.popoverLink[0];
+          $popoverLink[0] === this.popoverLink;
         if (!samePopover) {
           this.hidePopover();
-          this.showPopover(popoverLink);
+          this.showPopover($popoverLink);
         }
       } else if (this.popoverLink != null &&
-        !this.popoverContainsElement(event.target)) {
+        this.elementIsOutsidePopover(event.target)) {
         this.hidePopover();
       }
     },
+    actionsId(index) {
+      return `field-key-list-actions${index}`;
+    },
+    showRevoke(fieldKey) {
+      // Bootstrap does not actually disable dropdown menu items marked as
+      // disabled.
+      if (fieldKey.isRevoked()) return;
```

Comment:
i think in this case it's fine, but in more complex cases it'd be nice to have the logic for both conditions in the same place somewhere, so the ui and the business logic absolutely agree on when disabled should happen